source/game_sa/GameLogic.cpp

```diff
@@ -113,127 +142,388 @@ void CGameLogic::InitAtStartOfGame() {
+    }
+    auto playerData = player->m_pPlayerData;
+    auto playerInfo = player->GetPlayerInfoForThisPlayerPed();
+
+    player->physicalFlags.bDestroyed = false;
+    player->m_fArmour = 0.0f;
+    player->m_fHealth = static_cast<float>(playerInfo->m_nMaxHealth);
+    player->m_bIsVisible = true;
+    player->m_nDeathTime = 0;
+    player->bDoBloodyFootprints = false;
+    playerData->m_nDrunkenness = 0;
+    playerData->m_nFadeDrunkenness = 0;
+    CMBlur::ClearDrunkBlur();
+    playerData->m_nDrugLevel = 0;
+    player->ClearAdrenaline();
+    player->ResetSprintEnergy();
+    if (auto& fire = player->m_pFire) {
+        fire->createdByScript = false;
+        fire->Extinguish();
+        fire = nullptr;
+    }
+    player->m_pedAudio.TurnOffJetPack();
+    player->bInVehicle = false;
+    if (auto vehicle = player->m_pVehicle) {
+        vehicle->CleanUpOldReference((CEntity**)&vehicle);
+    }
+    player->m_pVehicle = nullptr;
+    player->GetWanted()->Reset();
+    player->RestartNonPartialAnims();
+    playerInfo->MakePlayerSafe(false, 10'000.0f);
+    player->m_bRemoveFromWorld = false;
+    player->ClearWeaponTarget();
+    player->SetInitialState(false);
+    player->GetAcquaintance() = CPedType::GetPedTypeAcquaintances(player->m_nPedType);
+    CCarCtrl::ClearInterestingVehicleList();
+
+    player->Teleport(posn + CVector{0.0f, 0.0f, 1.0f}, false);
+    player->m_fCurrentRotation = player->m_fAimingRotation = DegreesToRadians(playerStartHeading);
+    player->SetHeading(player->m_fCurrentRotation);
+
+    CTheScripts::ClearSpaceForMissionEntity(posn, player);
+    CWorld::ClearExcitingStuffFromArea(posn, 4000.0, 1);
+    player->RestoreHeadingRate();
+    player->m_nAreaCode = AREA_CODE_NORMAL_WORLD;
+    player->m_pEnex = 0;
+    CEntryExitManager::ms_entryExitStackPosn = 0;
+    CGame::currArea = AREA_CODE_NORMAL_WORLD;
+    CPopulation::bInPoliceStation = 0;
+    CStreaming::RemoveBuildingsNotInArea(AREA_CODE_NORMAL_WORLD);
+    TheCamera.SetCameraDirectlyBehindForFollowPed_CamOnAString();
+    TheCamera.RestoreWithJumpCut();
+    CReferences::RemoveReferencesToPlayer();
+    CGarages::PlayerArrestedOrDied();
+    CStats::SetStatValue(STAT_KILLS_SINCE_LAST_CHECKPOINT, 0.0);
+    CShopping::RemoveLoadedShop();
+    CWorld::Remove(player);
+    CWorld::Add(player);
+    CHud::ResetWastedText();
+    CStreaming::StreamZoneModels(posn);
+    CPostEffects::m_smokeyEnable = false;
+    CTimeCycle::StopExtraColour(0);
+    CPostEffects::ScriptResetForEffects();
+
+    if (auto clothesDesc = playerData->m_pPedClothesDesc; clothesDesc->m_anModelKeys[9]) {
+        clothesDesc->SetTextureAndModel(nullptr, nullptr, CLOTHES_TEXTURE_SPECIAL);
+        CClothes::RebuildPlayer(player, false);
+    } else {
+        CClothes::RebuildPlayerIfNeeded(player);
+    }
+    CAudioZones::Update(true, posn);
+    AudioEngine.StopPoliceScanner(true);
+    CWaterLevel::FindNearestWaterAndItsFlow();
+    TheCamera.m_fDistanceToWater = 99999.898f;
```

Comment:
```suggestion
    TheCamera.m_fDistanceToWater = 100'000.0f;
```

![image](https://user-images.githubusercontent.com/51059739/198831918-0bd06dfa-f600-4406-b2ec-548073f46daa.png)